lib/bootstrap.js

```diff
@@ -5,11 +5,14 @@ import net from 'net';
 import path from 'path';
 import _ from 'lodash';
 import { errorFromCode } from 'mobile-json-wire-protocol';
+import B from 'bluebird';
 
 class AndroidBootstrap {
   constructor (systemPort, webSocket) {
     this.systemPort = systemPort;
     this.webSocket = webSocket;
+    this.onUnexpectedShutdown = new B(() => {}).cancellable();
+    this.uiAutomatorUnexpectedShutdown = false;
```

Comment:
it is unclear to me why we need to keep track of this.

it seems like you envision a scenario where we get the unexpected uiautomator exit and _then_ call `this.shutdown`. but i don't see that ever happening...?